### PR TITLE
Follow rules-of-hooks lint rule

### LINKED ISSUE #3675: themedUseStyletron hook is not matched by the rules-of-hooks lint rule
We have some 'themed' styletron wrappers in baseweb for when developers add items to the theme and want to get that type propagated through to constructing styletron-components. These function names are `themedStyled`, `themedWithStyle`, and `themedUseStyletron`. As described in the title, `themedUseStyletron` is a react hook and must be prefixed with `use` to be matched by the lint rule.

A simple fix would be to move `themed` from prefix to suffix position.
`styledThemed`, `withStyleThemed`, `useStyletronThemed`

This should be updated within the baseweb documentation. There are also some constructor functions that should be aliased to better match the new naming.

`createThemedUseStyletron` => `createUseStyletronThemed`
`createThemedStyled` => `createStyledThemed`
`createThemedWithStyle` => `createWithStyleThemed`

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #3675 

#### Description
Renamed the following:

`themedStyled => styledThemed`
`themedWithStyle => withStyleThemed`
`themedUseStyletron => useStyletronThemed`
`createThemedUseStyletron => createUseStyletronThemed`
`createThemedStyled => createStyledThemed`
`createThemedWithStyle => createWithStyleThemed`

#### Scope
Minor: Linting

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {Locale} from './locale';
 import {Theme, ThemePrimitives} from './theme';
 
 type UseStyletronFn<Theme> = () => [(arg: StyleObject) => string, Theme];
-export function createThemedUseStyletron<Theme>(): UseStyletronFn<Theme>;
+export function createUseStyletronThemed<Theme>(): UseStyletronFn<Theme>;
```
Comment (chasestarr) [inline src/index.d.ts:13]: Please maintain the existing function declarations and create new aliases for the names

File: src/index.js [hunk 2]
```diff
@@ -10,9 +10,9 @@ export {
   withStyle,
   withWrapper,
   useStyletron,
-  createThemedStyled,
-  createThemedWithStyle,
-  createThemedUseStyletron,
+  createStyledThemed,
+  createWithStyleThemed,
+  createUseStyletronThemed,
```
Comment (chasestarr) [inline src/index.js:15]: Please maintain the existing exports

File: src/styles/index.js [hunk 3]
```diff
@@ -9,9 +9,9 @@ LICENSE file in the root directory of this source tree.
 import {ThemeContext} from './theme-provider.js';
 
 export {
-  createThemedStyled,
-  createThemedWithStyle,
-  createThemedUseStyletron,
+  createStyledThemed,
+  createWithStyleThemed,
+  createUseStyletronThemed,
```
Comment (chasestarr) [inline src/styles/index.js:14]: Please maintain the existing exports

**Review by chasestarr — COMMENTED** [review]
Thanks @im-adithya. I've made a few comments where I'd like to maintain the existing API. Please only make additions in this PR. I also would expect some flow type definition changes in some `types.js` files. Can you please try running `yarn flow` locally?

File: src/index.d.ts (hunk 1 above)
Comment (im-adithya) [reply src/index.d.ts:13]: Can you please let me know what exactly do you mean by aliasing?

**Comment by im-adithya** [comment]
@chasestarr made few changes regarding maintaining exports, please check.

> I also would expect some flow type definition changes in some `types.js` files. Can you please try running `yarn flow` locally?

I did but didn't get any error regarding `types.js` files, is there any issue with it?

**Comment by chasestarr** [comment]
Thank you for contributing this PR. I apologize, but landing this would require us to codemod a decent amount of internal code and we will not be prioritizing that effort.

**Pull request closed without merging**